packages/runtime/container-runtime/src/opCompressor.ts

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { IsoBuffer } from "@fluidframework/common-utils";
+import { IBatchMessage } from "@fluidframework/container-definitions";
+import { compress } from "lz4js";
+import { BatchMessage } from "./batchManager";
+import { CompressionAlgorithms, ContainerRuntimeMessage } from "./containerRuntime";
+
+/**
+ * Compresses batches of ops. It generates a single compressed op that contains
+ * the contents of each op in the batch. It then submits empty ops for each original
+ * op to reserve sequence numbers.
+ */
+export class OpCompressor {
+    private compressedBatchCount = 0;
+
+    constructor(private readonly logger: ITelemetryLogger) {}
+
+    public compressBatch(batch: BatchMessage[], originalLength: number): IBatchMessage[] {
+        const batchToSend: IBatchMessage[] = [];
+        this.compressedBatchCount++;
+        const batchedContents: ContainerRuntimeMessage[] = [];
+
+        for (const message of batch) {
+            batchedContents.push(message.deserializedContent);
+        }
+
+        const compressionStart = Date.now();
+        const contentsAsBuffer = new TextEncoder().encode(JSON.stringify(batchedContents));
+        const compressedContents = compress(contentsAsBuffer);
+        const compressedContent = IsoBuffer.from(compressedContents).toString("base64");
+        const duration = Date.now() - compressionStart;
+
+        if (this.compressedBatchCount % 100) {
```

Comment:
I'd say that the most likely usage will be - "please compress batches above 200K".
And if so, we can report data about every invocation.

That clearly could change in the future. I do not know how much we want to make it future prof, but it can be something - sample below 100K, report every one above 200K or something like that.

Basically, there is a value to report every instance if events do not happen often